backend/src/main/java/com/google/sps/utils/RecipeCollector.java

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.utils;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.sps.data.Recipe;
+import java.util.ArrayList;
+import java.util.List;
+
+public final class RecipeCollector {
+
+  public static List<Recipe> getRecipes(List<Long> recipeIds, DatastoreService datastore) {
+    List<Recipe> recipes = new ArrayList<>();
+    if (recipeIds == null) return recipes;
+    for (int i = 0; i < recipeIds.size(); i++) {
+      int recipeId = recipeIds.get(i).intValue();
+      Query query =
+          new Query(RecipeConstants.ENTITY_RECIPE)
+              .setFilter(
+                  new Query.FilterPredicate(
+                      RecipeConstants.PROPERTY_RECIPE_ID, Query.FilterOperator.EQUAL, recipeId));
+      PreparedQuery results = datastore.prepare(query);
+      Entity recipeEntity = results.asSingleEntity();
```

Comment:
I think it should be possible to make `Entity` and `Recipe`n in a way that you can create a new `Recipe` without having to manually populate the fields.

I found this from a quick search: https://docs.oracle.com/cd/E19159-01/819-3669/bnbro/index.html

Reply:
I've tried using objectify (https://github.com/objectify/objectify) to do that, but I couldn't make it work because I had issues with the local setup. I can look into that again in a week or so